### ISSUE TITLE
More lightweight interactions

### ISSUE THREAD

**Issue by sblom** [upperstory/spintronics-simulator#1]
The currently deployed simulator requires an awful lot of clicking to build a circuit (choose a part, place the part, choose a different part or the interact tool, etc.).

I grabbed the code in the `working` branch, and noticed that it's already much better due to defaulting to the Interact tool after I place a part.

It feels like this could go one step further by allowing dragging parts (currently, the ghost part stops moving when I release my click (or touch), but it doesn't get dropped. If click, click to choose, place is desired, it's possible to support both at the same time.

If hovering over a part highlighted it, it would be easy to enable keyboard interactions as well, such as `Del` to delete, `m` to move, etc.

I don't have a complete and coherent scheme for what the interactions should feel like, but examples of CAD-like web experiences that feel really clean include Sketchup (there's a free tier), or TinkerCAD (their circuit editor feels like something to aspire to here).

For that matter, partnering with something like TinkerCAD where you can share circuits and edit them slightly would be amazing, and is probably part of the vision here.

**Bottom line for now**, would you mind if I tried contributing a Pull Request for something simple like drag-and-drop from the left to place components (while still allowing click,click to place components also)?

PS I've got all of the perspectives (at various times in my life) of developer, startup CTO, and open source maintainer, so please don't take any of this as brash criticism--I'm open to feedback on how to contribute (or not) in whatever way(s) are most helpful.
